networkit/cpp/generators/BarabasiAlbertGenerator.cpp

```diff
@@ -84,20 +84,18 @@ Graph BarabasiAlbertGenerator::generateOriginal() {
         while (targets.size() - 1 < k) {
             uint64_t random = (uint64_t) Aux::Random::integer(degreeSum);
             j++;
-            ///if (j > k) throw std::runtime_error("Possible infinite loop detected.");
             bool found = false; // break from node iteration when done
             auto notFound = [&](){ return ! found; };
 
             G.forNodesWhile(notFound, [&](node v) {
-
-
                 if (random <= G.degree(v)) {
                     found = true; // found a node to connect to
                     targets.insert(v);
                 }
                 random -= G.degree(v);
-                //if (j >= G.numberOfNodes() && found==false) throw std::runtime_error("Last node, but still nothing happened.");
             });
+
+            handler.assureRunning();
         }
 
         targets.erase(u);
```

Comment:
Is it necessary to add edges in a specific order? Otherwise, would an `unordered_set` help?